### PR TITLE
Create MutinyDLCManager

### PR DESCRIPTION
**Pull request by benthecarman**
build is failing for me from some dependency issue, putting up for now though

### CODE REVIEW

File: node-manager/src/dlc.rs [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+        let xkey: ExtendedKey = ExtendedKey::from(xprivkey);
+        let master_xprv = xkey.into_xprv(network).unwrap();
+        let coin_type = match network {
+            Network::Bitcoin => 0,
+            Network::Testnet => 1,
+            Network::Signet => 1,
+            Network::Regtest => 1,
+        };
+
+        let base_path = DerivationPath::from_str("m/585'").unwrap();
```
Comment (TonyGiorgio) [inline node-manager/src/dlc.rs:48]: we should try to keep track of these derivation paths and where they are use. Phantom also uses a shared derivation path as well. And some of the node derivation paths should probably be hard coded so they aren't used either.

File: node-manager/src/dlc.rs [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+        let network = self.wallet.network();
+        let xprivkey = ExtendedPrivKey::new_master(network, &entropy).unwrap();
+        let xkey: ExtendedKey = ExtendedKey::from(xprivkey);
+        let master_xprv = xkey.into_xprv(network).unwrap();
+        let coin_type = match network {
+            Network::Bitcoin => 0,
+            Network::Testnet => 1,
+            Network::Signet => 1,
+            Network::Regtest => 1,
+        };
```
Comment (TonyGiorgio) [inline node-manager/src/dlc.rs:46]: Weird that you had to do this, I put some of this in `utils` though if you want to move it there though. No worries if not.

File: node-manager/src/dlc.rs [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+}
+
+impl dlc_manager::Signer for MutinyDLCWallet {
+    fn sign_tx_input(
+        &self,
+        tx: &mut bitcoin::blockdata::transaction::Transaction,
+        input_index: usize,
+        tx_out: &bitcoin::blockdata::transaction::TxOut,
+        redeem_script: Option<bitcoin::blockdata::script::Script>,
+    ) -> Result<(), ManagerError> {
```
Comment (TonyGiorgio) [inline node-manager/src/dlc.rs:70]: Now that we've got multiple "wallets", maybe putting this inside some generic BDK wallet wrapper could be useful. Maybe not now but having multiple "wallets" and "chains" can get rather confusing and probably some leakage in terms of reusing logic. I don't think I see anything DLC specific here.

File: node-manager/src/dlc.rs [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+            .database()
+            .get_sync_time()
+            .map_err(bdk_err_to_manager_err)?
+            .unwrap();
+
+        Ok(last_sync.block_time.height as u64)
+    }
+
+    fn get_block_at_height(&self, _height: u64) -> Result<Option<Block>, ManagerError> {
+        Ok(None)
```
Comment (TonyGiorgio) [inline node-manager/src/dlc.rs:273]: did you not need to add async functions after all? Is it not actually needed?

**Review by TonyGiorgio — COMMENTED** [review]
nothing terribly needed to change but some comments. feel free to skip if you want

File: node-manager/src/dlc.rs (hunk 3 above)
Comment (benthecarman) [reply node-manager/src/dlc.rs:70]: yeah this is just kinda a hack to sign a specific input, but I agree that things could probably be cleaned up. That is kinda what I want to propose for the spiral grant lol

File: node-manager/src/dlc.rs (hunk 4 above)
Comment (benthecarman) [reply node-manager/src/dlc.rs:273]: Yeah I just changed it to `Option` instead, was simpler: https://github.com/benthecarman/rust-dlc/commit/c13700440c7a5aeb8edd9ce27cfc3e1c36a92217

This is only needed for the DLC channels stuff so I we'd have to punt on that part for now.

**Pull request merged**